ctru-rs/src/services/cam.rs

```diff
@@ -0,0 +1,1050 @@
+        &self,
+        camera: CamSelect,
+        white_balance: CamWhiteBalance,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetWhiteBalance(camera.bits(), white_balance.bits());
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn set_white_balance_without_base_up(
+        &self,
+        camera: CamSelect,
+        white_balance: CamWhiteBalance,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r =
+                ctru_sys::CAMU_SetWhiteBalanceWithoutBaseUp(camera.bits(), white_balance.bits());
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn set_sharpness(&self, camera: CamSelect, sharpness: i8) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetSharpness(camera.bits(), sharpness);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn set_auto_exposure(&self, camera: CamSelect, enabled: bool) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetAutoExposure(camera.bits(), enabled);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn is_auto_exposure(&self, camera: CamSelect) -> crate::Result<bool> {
+        unsafe {
+            let mut is_auto_exposure = false;
+            let r = ctru_sys::CAMU_IsAutoExposure(&mut is_auto_exposure, camera.bits());
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(is_auto_exposure)
+            }
+        }
+    }
+
+    pub fn set_auto_white_balance(&self, camera: CamSelect, enabled: bool) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetAutoWhiteBalance(camera.bits(), enabled);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn is_auto_white_balance(&self, camera: CamSelect) -> crate::Result<bool> {
+        unsafe {
+            let mut is_auto_white_balance = false;
+            let r = ctru_sys::CAMU_IsAutoWhiteBalance(&mut is_auto_white_balance, camera.bits());
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(is_auto_white_balance)
+            }
+        }
+    }
+
+    pub fn flip_image(
+        &self,
+        camera: CamSelect,
+        flip: CamFlip,
+        context: CamContext,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_FlipImage(camera.bits(), flip.bits(), context.bits());
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn set_detail_size(
```

Comment:
Clippy doesn't like this. :paperclip:
Maybe use tuples for x/y pairs, or ignore the lint.
```
warning: this function has too many arguments (9/7)
   --> ctru-rs/src/services/cam.rs:641:5
    |
641 | /     pub fn set_detail_size(
642 | |         &self,
643 | |         camera: CamSelect,
644 | |         width: i16,
...   |
650 | |         context: CamContext,
651 | |     ) -> crate::Result<()> {
    | |__________________________^
    |
    = note: `#[warn(clippy::too_many_arguments)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments
```

Reply:
Related to my `CamContext` comments.